integration-tests/src/tests/runtime/sanity_checks.rs

```diff
@@ -0,0 +1,453 @@
+    "test_contract.alice.near".parse().unwrap()
+}
+
+fn setup_runtime_node_with_contract(wasm_binary: &[u8]) -> RuntimeNode {
+    // Create a `RuntimeNode`. Load `RuntimeConfig` from `RuntimeConfigStore`
+    // to ensure we are using the latest configuration.
+    let mut genesis =
+        Genesis::test(vec![alice_account(), bob_account(), "carol.near".parse().unwrap()], 3);
+    add_test_contract(&mut genesis, &alice_account());
+    add_test_contract(&mut genesis, &bob_account());
+    let runtime_config_store = RuntimeConfigStore::new(None);
+    let runtime_config = runtime_config_store.get_config(PROTOCOL_VERSION);
+    let node = RuntimeNode::new_from_genesis_and_config(
+        &alice_account(),
+        genesis,
+        RuntimeConfig::clone(runtime_config),
+    );
+
+    let account_id = node.account_id().unwrap();
+    let node_user = node.user();
+    let tx_result = node_user
+        .create_account(
+            account_id,
+            test_contract_account(),
+            node.signer().public_key(),
+            TESTING_INIT_BALANCE / 2,
+        )
+        .unwrap();
+    assert_eq!(tx_result.status, FinalExecutionStatus::SuccessValue(to_base64(&[])));
+    assert_eq!(tx_result.receipts_outcome.len(), 2);
+
+    let tx_result =
+        node_user.deploy_contract(test_contract_account(), wasm_binary.to_vec()).unwrap();
+    assert_eq!(tx_result.status, FinalExecutionStatus::SuccessValue(to_base64(&[])));
+    assert_eq!(tx_result.receipts_outcome.len(), 1);
+
+    node
+}
+
+/// Calls method `sanity_check` on `test-contract-rs` and verifies that the
+/// resulting gas profile matches expectations.
+///
+/// This test intends to catch accidental configuration changes, see #4961.
+#[test]
+fn test_cost_sanity() {
+    let test_contract = if cfg!(feature = "nightly_protocol") {
+        near_test_contracts::nightly_rs_contract()
+    } else {
+        near_test_contracts::rs_contract()
+    };
+    let node = setup_runtime_node_with_contract(test_contract);
+    let data = serde_json::json!({
+        "contract_code": to_base64(near_test_contracts::trivial_contract()),
+        "method_name": "main",
+        "method_args": to_base64(&[]),
+        "validator_id": bob_account().as_str(),
+    });
+    let res = node
+        .user()
+        .function_call(
+            alice_account(),
+            test_contract_account(),
+            "sanity_check",
+            serde_json::to_vec(&data).unwrap(),
+            MAX_GAS,
+            0,
+        )
+        .unwrap();
+    assert_eq!(res.status, FinalExecutionStatus::SuccessValue(to_base64(&[])));
+    assert_eq!(res.transaction_outcome.outcome.metadata.gas_profile, None);
+    assert_receipts_status(
+        "sanity_check",
+        res.receipts_outcome.clone(),
+        expected_cost_sanity_receipts_statuses(),
+    );
+    assert_gas_profiles(
+        "sanity_check",
+        res.receipts_outcome.clone(),
+        expected_cost_sanity_gas_profiles(),
+    );
```

Comment:
Let's use https://github.com/mitsuhiko/insta here (which we already use elswhere) to avoid writing a boilerplate comparison function